src/hotspot/share/cds/cdsHeapVerifier.cpp

```diff
@@ -232,10 +265,19 @@ inline bool CDSHeapVerifier::do_entry(oop& orig_obj, HeapShared::CachedOopInfo&
       // should be flagged by CDSHeapVerifier.
       return true; /* keep on iterating */
     }
+    if (info->_holder->is_hidden()) {
+      return true;
+    }
     ResourceMark rm;
+    char* class_name = info->_holder->name()->as_C_string();
+    char* field_name = info->_name->as_C_string();
+    if (strstr(class_name, "java/lang/invoke/BoundMethodHandle$Species_") == class_name &&
```

Comment:
Can we have a comment to explain this special case?

Reply:
This check is no longer necessary, the Species_XXX classes are now added to the aot-init list in `AOTClassInitializer::can_archive_initialized_mirror()`. I've removed these two lines.